### PR TITLE
Issue 907 metabolink mergeprep

### PR DESCRIPTION
**Pull request by turbomam**
built on an existing branch from @JamesTessmer  and @anastasiyaprymolenna, which addresses
- #907 
but back-merges the current main

### CODE REVIEW

**Comment by turbomam** [comment]
I will be using `gen-linkml src/schema/SampleOperations.yaml ` to check the soundness of the new schema file

> ValueError: Slot: permissible_values - attribute text value (File "SampleOperations.yaml", line 279, col 15: screw_top_concial tube) does not match key (screw_top_conical)

You don't have to provide a text for a permissible value. In fact, I recommend leaving it out. If you do want to provide a `text`,it has to match the key in the YAML file. If you want to provide elaboration, you can use many other [attributes allowed for PermissibleValues](https://linkml.io/linkml-model/docs/PermissibleValue/)

- aliases (list of terms from some other recognized controlled vocabulary, or terms well-known to people famiar wiht teh domain of this subject matter)
- comments (list of strings)
- description (string)
- meaning (a CURIe for a ontology class, preferably from EBI OLS)
  - you could use [OBI:0000836](https://www.ebi.ac.uk/ols4/ontologies/obi/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FOBI_0000836?lang=en) (test tube) here, but that's probably overly broad.
- title (string)

You describe your classes like  `labDevice` as "labDevice schema". From a LinkML perspective, there will only be one schema in this repo, the nmdc-schema. I don't see how `labDevice` is any different from any other class. How disruptive you it be to you to remove these descriptions? Or replace them with something that tells an intelligent but unfamiliar person what they are? Like isn't a labDevice a 

> A material entity that is designed to perform a function in a scientific investigation, but is not a reagent.
from ["device" in the Ontology of Biomedical Investigations](https://www.ebi.ac.uk/ols4/ontologies/obi/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FOBI_0000968)?

Why is there a `description **` slot in  `labDevice`? I have removed the asterisks.

Here are some reports from `linkml-lint src/schema/SampleOperations.yaml`

> warning  Class has name 'labDevice'  (standard_naming)
> warning  Enum has name 'devicetypeenum'  (standard_naming)
>  warning  Slot has name 'Site_Code'  (standard_naming) (and many more complaints about PascalCase slot names)

I would like for the `Activity` enumeration to be called `ActivityEnum`, because we already have an element in the schema called `Activity`

The linter recommends lower_snake_case for permissible values. I'd like to stick with that here, and impose it on legacy parts of the schema in the future. 

ActivityEnum has one permissible value that doesn't seem to fit in: "AnalyticalSample". That sounds like a material entity, not an activity. Can you elaborate on that?

The linter wants all elements to have descriptions. Your missing several for slots and enums. Let's work on that together.

Are you sure that the `xsd` prefix needs to be defined here? I believe it's asserted in nmdc.yaml, so probably not. If you do want to assert it, let's be consistent and use the more compact form

```yaml
xsd: http://www.w3.org/2001/XMLSchema#
```

I don't see any reason why nmdc.yaml or core.yaml should be imported into this module. If they were needed, they should appear in the same case as the imported file, like nmdc, not NMDC.

I would like for us to consolidate the Process, Activity and Operations terminology used throughout the schema. If you would like to retain the word operation(s) here, lets put that in aliases, descriptions or comments.

**Comment by JamesTessmer** [comment]
Thanks for the feedback Mark. A fair number of those are just issues from me not knowing best practices of LinkML so I appreciate you pointing those out.

You mentioned the '____ schema' descriptions which I used as kind of a filler description for ones that didn't have a description in the initial code Anastasiya gave me, so those should be fine to remove.

I'm still relatively new the LinkML so I thought the XSD had to be defined in every file, and since we were using the biosample and quantityValue definitions the files containing those would need to be imported. Does LinkML share the definitions across files?

One other thing I wasn't sure about but would like to point out is in line 2 for the ID, I currently have a temporary MONet id. (that I'm not sure actually points to anything)  Does this need to be replaced with an NMDC id?


I can make the other changes I'm capable of such as lower snake case, unless you were planning to do that.

**Comment by JamesTessmer** [comment]
Also, another question I should've put in my last comment: Is this PR going to be the one that gets merged once everything is ready or is this just to figure out changes we need and then we will merge the other 907 branch?

**Comment by turbomam** [comment]
I'm going to contact you by Slack now but we can return to these PR notes any times.

**Comment by turbomam** [comment]
You have redefined at least the `AnalyticalSample` class, relative to the rest of the schema. We generally refer to classes, slots, etc by bare name, so the fact that they could have different prefixes isn't enough unless we mentions them with prefix:name.

**Review by turbomam — COMMENTED** [review]
Only JSON, or preferably YAML files should go in src/data

so move src/data/invalid/MPLEx paper.pdf somewhere else... I'm not even sure where it would belong in this repo

**Review by turbomam — COMMENTED** [review]
so move src/data/invalid/MPLEx paper.pdf somewhere else... I'm not even sure where it would belong in this repo

**Comment by turbomam** [comment]
Good work, and I'm looking forward to our meeting today.

In addition to the unresolved conflicts reported on this page, I can't build this branch

> ValueError: File "sample_operations.yaml", line 83, col 9 Class "MicrobialBiomassDetermination" - unknown slot: "extraction_operation"

**Comment by turbomam** [comment]
We should clean up the schema files:

- src/modules-not-in-use
  - src/modules-not-in-use/sample_operations.yaml
  - src/modules-not-in-use/sample_prep.yaml
- src/schema
  - src/schema/sample_operations.yaml
  - src/schema/sample_prep.yaml

If we're certain that there's nothing useful left in  `src/modules-not-in-use/sample_operations.yaml` and `src/modules-not-in-use/sample_prep.yaml`, then we should delete the `src/modules-not-in-use/ `directory

Which of `src/modules-not-in-use` and `src/schema/sample_prep.yaml` are you actively maintaining? Whichever that is, do you need anything from the other:? If not,let's delete it.

**Comment by JamesTessmer** [comment]
> We should clean up the schema files:
> 
> * src/modules-not-in-use
>   
>   * src/modules-not-in-use/sample_operations.yaml
>   * src/modules-not-in-use/sample_prep.yaml
> * src/schema
>   
>   * src/schema/sample_operations.yaml
>   * src/schema/sample_prep.yaml
> 
> If we're certain that there's nothing useful left in `src/modules-not-in-use/sample_operations.yaml` and `src/modules-not-in-use/sample_prep.yaml`, then we should delete the `src/modules-not-in-use/ `directory
> 
> Which of `src/modules-not-in-use` and `src/schema/sample_prep.yaml` are you actively maintaining? Whichever that is, do you need anything from the other:? If not,let's delete it.

sample_prep isn't one that we made for this issue/branch, I think it was an earlier start at doing what we're doing now. I think Brynn is in the process of figuring out what is/isn't being used by anything else from that file before we make changes to it so that we don't accidentally break something.

File: src/data/valid/Database-microbialbiomass.yaml [hunk 1]
```diff
@@ -0,0 +1,301 @@
+protocol_execution_set:
+  - id: generic:Campaign_1
+    name: Microbial Biomass # does this need to be a unique name to the execution?
```
[the remark this answers is not in the document]
Comment (mslarae13) [reply src/data/valid/Database-microbialbiomass.yaml:3]: @JamesTessmer

**Review by mslarae13 — COMMENTED** [review]
@JamesTessmer see comments and lets discuss

**Pull request closed without merging**